login-module/src/core.js

```diff
@@ -0,0 +1,99 @@
+import { WebAuth } from 'auth0-js'
+import Quiz from './quiz'
+import Enrollment from './enrollment'
+import Certificate from './certificate'
+
+function validateOptions(options, requiredOptions, defaultOptions) {
+	const opts = { ...defaultOptions, ...options }
+	const hasRequiredOptions = requiredOptions.every(item => opts[item])
+	if (!hasRequiredOptions) {
+		throw new Error(`required option missing - one of ${requiredOptions.join(', ')}`)
+	}
+	return opts
+}
+
+class GraphAcademyServices {
```

Comment:
This class can access authenticated services. 
The `authResult` is mandatory. For extra safety, we could also check that `accessToken` is defined. 
We can then safely assume that `this.authResult.accessToken` will return a valid access token.